src/components/OrgFile/components/Header/index.js

```diff
@@ -305,6 +308,26 @@ ${header.get('rawDescription')}`;
     // window.location.href = mailtoURI;
   }
 
+  handleAddNoteClick() {
+    let input = prompt('Enter a note to add to the header:');
+    if (input !== null) input = input.trim();
+    if (!input) return;
+    const { header } = this.props;
+
+    const dontIndent = this.props.dontIndent;
+
+    // Getting indentation, timestamp, and template string is impure, hence do it here:
+    const indentation = dontIndent ? '' : ' '.repeat(header.nestingLevel + 1);
```

Comment:
@munen I'm not sure if I got your previous comment right (on composing the note in the reducer). Anyway, getting the indentation, template string and the timestamp is impure code and the last step, filling the template string can be done right here, I think.

Reply:
> Anyway, getting the indentation, template string and the timestamp is impure code and the last step, filling the template string can be done right here, I think.

You can do it right here. But then it's harder to test and you have coupling to the component.

Also all the things that you mentioned shouldn't be impure unless I misunderstand something.

In any case, it's not a big issue. You can do it in the component and when the PR is done, and if I'm right that the things you brought up are in fact pure, then I can easily refactor it into the reducer before merging the PR. I'll happily take over that part.

Reply:
@schoettl You are correct that changes to the state should only be made with pure functions. That is the third of the [three principles of Redux](https://redux.js.org/introduction/three-principles).

However, that does not include a query for `dontIndent`. The reason is that this information is in the store - and the store is the single source of truth. That's the first principle of Redux. Think of it like that: It is as if the state was an implicit input to the function you're writing.

Creating the `noteText` should be done in the reducer. Otherwise the functionality of this feature would be tightly coupled to the component in which it is happening. However, components shouldn't hold this kind of business logic for multiple reasons. One: It's not possible to test the functionality outside the scope of a fully hydrated component.

But as I mentioned before: If this is unclear, you can totally build the `noteText` in the component. And when you're done, I'll refactor it to the Reducer and maybe add a test(;

Reply:
@munen 
Great, totally clear. I'll pass just the time and input text to the reducer. However, I already refactored and reuse the current `addNote` function from another reducer with a different text (the log when repeating tasks are done). I'll refactor this...

I also noticed that "get current time" is used in some other reducers. I'd refactor this to pass the time from the component, do you agree?

Reply:
>  Great, totally clear. I'll pass just the time and input text to the reducer. However, I already refactored and reuse the current addNote function from another reducer with a different text (the log when repeating tasks are done). I'll refactor this...

:+1: 

> I also noticed that "get current time" is used in some other reducers. I'd refactor this to pass the time from the component, do you agree?

That sounds like a great idea! It'll make testing them easier, too. Otherwise the only option to reliably test them are:

- Mock time
- Use Regexps to test for 'time and dates' in the results instead of a specific time and date

:pray: